internal/monitors/haproxy/metadata.yaml

```diff
@@ -0,0 +1,617 @@
+monitors:
+  - monitorType: haproxy
+    doc: |
+      This monitor scrapes [HAProxy](http://www.haproxy.org/) statistics from an HTTP endpoint or a UNIX socket.
+      It requires HAProxy 1.8+ and supports running HAProxy in multi-process mode where the HAProxy process each
+      metric pertains to is indicated by the metric dimension `process_num`.
+
+      <!--- SETUP --->
+      ### HTTP Endpoint Config
+      HAProxy stats must be enabled as described [here](https://www.haproxy.com/blog/exploring-the-haproxy-stats-page/).
+      Define the HAProxy monitor in the agent configuration file and provide the `csv export` URL where HAProxy
+      stats are served in CSV format. This monitor supports basic HTTP authentication. Simply provide the
+      username and password if required as shown below:
+      ```
+      ...
+      monitors:
+        - type: haproxy
+          url: "http://localhost:8404/stats?stats;csv"
+          username: "your username here"
+          password: "your password here"
+      ...
+      ```
+      Note: The HAProxy process level stats given by the [show info](https://cbonte.github.io/haproxy-dconv/1.8/management.html#9.3-show%20info) command
+      are not available at the HTTP endpoint. Scrape stats from the UNIX socket instead to include the `show info`
+      command metrics.
+
+      ### UNIX Socket Config
+      The location of the HAProxy socket file is defined in the HAProxy config file, as in the following example:
+      ```
+      global
+          daemon
+          stats socket /var/run/haproxy.sock
+          stats timeout 2m
+      ```
+      For the given location of the socket file, configure the HAProxy monitor URL as shown below:
+      ```
+      ...
+      monitors:
+        - type: haproxy
+          url: "unix:/var/run/haproxy.sock"
+      ...
+      ```
+      Note: The agent process needs read/write permissions to the HAProxy socket file.
+    dimensions:
+      proxy_name:
+        description: HAProxy configured proxy name. Possible values for listeners, frontends, backends, and servers.
+      service_name:
+        description: HAProxy configured service name. Possible values are FRONTEND for frontend, BACKEND for backend, configured name for server/listener.
+      process_num:
+        description: A number assigned to an HAProxy process instance (0 for first instance, 1 for second, ...).
+    metrics:
+      gauge.queue_current:
+        default: true
+        description: 'Number of current queued requests. For the backend this reports the number queued without
+        a server assigned. Values reported for backends and servers.'
+        type: gauge
+        properties:
```

Comment:
Properties don't exist on metrics, only on dimensions.  What is this trying to convey?

Reply:
removed